src/utilcode/winwrappers.cpp

```diff
@@ -0,0 +1,996 @@
+        {
+            ret = CopyFileW(
+                    Existingpath.GetUnicode(),
+                    Newpath.GetUnicode(),
+                    bFailIfExists
+                    );
+ 
+            lastError = GetLastError();
+        }
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+
+
+BOOL
+MoveFileWrapper(
+        _In_ LPCWSTR lpExistingFileName,
+        _In_ LPCWSTR lpNewFileName
+        )
+{
+    HRESULT hr  = S_OK;
+    BOOL    ret = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        Win32StackSString Existingpath(Win32StackSString::Literal, lpExistingFileName);
+        Win32StackSString Newpath(Win32StackSString::Literal, lpNewFileName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(Existingpath)) && SUCCEEDED(LongFile::NormalizePath(Newpath)))
+        {
+            ret = MoveFileW(
+                    Existingpath.GetUnicode(),
+                    Newpath.GetUnicode()
+                    );
+            
+            lastError = GetLastError();
+        }
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+
+BOOL
+MoveFileExWrapper(
+        _In_     LPCWSTR lpExistingFileName,
+        _In_opt_ LPCWSTR lpNewFileName,
+        _In_     DWORD    dwFlags
+        )
+{
+    HRESULT hr  = S_OK;
+    BOOL    ret = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        Win32StackSString Existingpath(Win32StackSString::Literal, lpExistingFileName);
+        Win32StackSString Newpath(Win32StackSString::Literal, lpNewFileName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(Existingpath)) && SUCCEEDED(LongFile::NormalizePath(Newpath)))
+        {
+            ret = MoveFileExW(
+                    Existingpath.GetUnicode(),
+                    Newpath.GetUnicode(),
+                    dwFlags
+                    );
+
+            lastError = GetLastError();
+        }
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+
+}
+
+DWORD
+SearchPathWrapper(
+        _In_opt_ LPCWSTR lpPath,
+        _In_ LPCWSTR lpFileName,
+        _In_opt_ LPCWSTR lpExtension,
+        _In_ BOOL getPath,
+        PathWCharString& lpBuffer,
+        _Out_opt_ LPWSTR * lpFilePart
+        )
+{
+    DWORD size;
+    HRESULT hr  = S_OK;
+    BOOL    ret = FALSE;
+    EX_TRY
+    {
+        Win32StackSString Existingpath(Win32StackSString::Literal, lpPath);
+        if (lpPath != NULL)
+        {
+            if (FAILED(LongFile::NormalizePath(Existingpath)))
+            {
+                ret = FALSE;
+            }
+            else
+            {
+                lpPath = Existingpath.GetUnicode();
```

Comment:
After this assignment, `lpPath` becomes a dangling pointer as soon as `Existingpath` goes out of scope.